--- mediastreamer-2.12.1/src/videofilters/msv4l2.c.orig	2015-10-19 13:42:10.603994267 +0200
+++ mediastreamer-2.12.1/src/videofilters/msv4l2.c	2016-03-31 18:44:22.923935372 +0200
@@ -343,7 +343,7 @@
 
 static int set_camera_feature(V4l2State *s, unsigned int ctl_id, int value, const char *feature_name){
 	struct v4l2_ext_control ctl={0};
-	struct v4l2_ext_controls ctls={0};
+	struct v4l2_ext_controls ctls={{0}};
 	struct v4l2_queryctrl queryctrl={0};
 	
 	queryctrl.id = ctl_id;
