--- mediastreamer2-4.5.22/src/audiofilters/pulseaudio.c.orig	2021-06-24 23:55:04.000000000 +0200
+++ mediastreamer2-4.5.22/src/audiofilters/pulseaudio.c	2021-07-18 10:10:08.244595475 +0200
@@ -229,7 +229,7 @@
  * output: sink and bidirectionnal cards into sink list, source only card into source list
  * merging is based on pulse audio card description */
 int pulse_card_compare(pa_device_t *sink, pa_device_t *source) {
-	return strncmp(sink->description, source->description, 512);
+	return strncmp(sink->description, source->description, PA_STRING_SIZE);
 }
 
 static void pulse_card_merge_lists(pa_device_t *pa_device, bctbx_list_t **pa_source_list) {
