--- mediastreamer2-5.4.17/CMakeLists.txt.orig	2025-05-19 20:19:27.509061004 +0200
+++ mediastreamer2-5.4.17/CMakeLists.txt	2025-05-19 20:23:06.907872418 +0200
@@ -321,7 +321,13 @@ if(NOT APPLE AND NOT ANDROID AND OPENGL_
 		set(MS2_USE_OLD_OPENGL_PROTOTYPE TRUE) # Before 1.11, Glew have few different headers from OpenGL Specs
 	endif()
 endif()
-# TODO: Handle SDL option
+if(ENABLE_SDL)
+	find_package(SDL)
+	if(NOT SDL_FOUND)
+		message(WARNING "Could not find SDL library")
+		set(ENABLE_SDL OFF CACHE BOOL "Enable SDL video output." FORCE)
+	endif()
+endif()
 if(ENABLE_THEORA)
 	find_package(Theora)
 	if(NOT Theora_FOUND)
@@ -522,6 +528,9 @@ endif()
 if(GLEW_FOUND)
 	list(APPEND LINK_LIBS GLEW::glew)
 endif()
+if(SDL_FOUND)
+	list(APPEND LINK_LIBS SDL)
+endif()
 if(Theora_FOUND)
 	list(APPEND LINK_LIBS ${Theora_TARGET})
 endif()
