diff -dur -x '*~' mediastreamer-2.9.0.orig/src/Makefile.am mediastreamer-2.9.0/src/Makefile.am
--- mediastreamer-2.9.0.orig/src/Makefile.am	2013-05-30 10:09:06.000000000 +0200
+++ mediastreamer-2.9.0/src/Makefile.am	2013-06-17 10:35:37.000000000 +0200
@@ -363,7 +363,7 @@
 
 AM_OBJCFLAGS=$(AM_CFLAGS)
 
-imgdir=$(datadir)/images/
+imgdir=$(datadir)/pixmaps/mediastreamer
 
 img_DATA=voip/nowebcamCIF.jpg
 
diff -dur -x '*~' mediastreamer-2.9.0.orig/src/videofilters/nowebcam.c mediastreamer-2.9.0/src/videofilters/nowebcam.c
--- mediastreamer-2.9.0.orig/src/videofilters/nowebcam.c	2013-02-20 14:38:36.000000000 +0100
+++ mediastreamer-2.9.0/src/videofilters/nowebcam.c	2013-06-17 10:37:20.000000000 +0200
@@ -1783,7 +1783,7 @@
 
 static char *def_image=NULL;
 
-static const char *def_image_path=PACKAGE_DATA_DIR "/images/" NOWEBCAM_JPG ".jpg";
+static const char *def_image_path="/usr/share/pixmaps/mediastreamer/" NOWEBCAM_JPG ".jpg";
 
 
 mblk_t *ms_load_nowebcam(MSVideoSize *reqsize, int idx){
