--- mediastreamer2-5.1.72/CMakeLists.txt.orig	2023-04-10 19:00:46.387121654 +0200
+++ mediastreamer2-5.1.72/CMakeLists.txt	2023-04-10 19:04:39.545858525 +0200
@@ -91,6 +91,8 @@ option(ENABLE_OPENGL_PROFILING "Measure
 option(ENABLE_SRTP "Build with the SRTP transport support." YES)
 cmake_dependent_option(ENABLE_ZRTP "Build with ZRTP support." YES "ENABLE_SRTP" NO)
 
+option(ENABLE_UPNP "Build with UPnP support." YES)
+
 option(ENABLE_SOUND "Can be used to turn off all possible sound backends." YES)
 cmake_dependent_option(ENABLE_ALSA "Enable ALSA support." YES "ENABLE_SOUND;LINUX_OR_BSD" NO)
 cmake_dependent_option(ENABLE_ANDROIDSND "Enable Android sound support." NO "ENABLE_SOUND;ANDROID" NO)
@@ -183,6 +185,13 @@ endif()
 if(ENABLE_PCAP)
 	find_package(PCAP QUIET)
 endif()
+if(ENABLE_UPNP)
+	find_library(UPNP upnp)
+	if(NOT UPNP)
+		message(WARNING "Could not find UPNP library.")
+		set(ENABLE_UPNP OFF CACHE BOOL "Build with UPnP support." FORCE)
+	endif()
+endif()
 if(ENABLE_SRTP)
 	if(NOT DISABLE_SRTP_SEARCH)
 		find_package(SRTP "2" REQUIRED)
@@ -449,6 +458,11 @@ endif()
 if(HAVE_DLOPEN)
 	list(APPEND LINK_LIBS dl)
 endif()
+if(ENABLE_UPNP)
+	list(APPEND LINK_LIBS upnp ixml)
+	list(APPEND MEDIASTREAMER2_INCLUDE_DIRS /usr/include/upnp)
+	add_definitions(-D_GNU_SOURCE)
+endif()
 if(ALSA_FOUND)
 	list(APPEND LINK_LIBS asound)
 	list(APPEND MEDIASTREAMER2_INCLUDE_DIRS ${ALSA_INCLUDE_DIRS})
--- mediastreamer2-4.5.22/src/CMakeLists.txt.orig	2021-07-09 21:45:54.498880091 +0200
+++ mediastreamer2-4.5.22/src/CMakeLists.txt	2021-07-09 21:45:58.885522993 +0200
@@ -140,6 +140,15 @@ if (APPLE)
         list(APPEND VOIP_SOURCE_FILES_OBJC utils/apple_utils.h utils/apple_utils.m)
 endif()
 
+if(ENABLE_UPNP)
+	list(APPEND VOIP_SOURCE_FILES_C
+		upnp/upnp_igd.c
+		upnp/upnp_igd_private.h
+		upnp/upnp_igd_cmd.c
+		upnp/upnp_igd_utils.c
+		upnp/upnp_igd_utils.h
+	)
+endif()
 if(ENABLE_ALSA)
 	list(APPEND VOIP_SOURCE_FILES_C audiofilters/alsa.c)
 endif()
