--- mediastreamer2-4.5.22/CMakeLists.txt.orig	2021-06-24 23:55:04.000000000 +0200
+++ mediastreamer2-4.5.22/CMakeLists.txt	2021-07-07 06:15:26.781790314 +0200
@@ -371,7 +371,7 @@ include_directories(
 )
 
 set(MEDIASTREAMER2_INCLUDE_DIRS )
-set(LINK_LIBS bctoolbox ortp)
+set(LINK_LIBS bctoolbox ortp pthread)
 
 if(WIN32)
 	if(NOT CMAKE_SYSTEM_NAME STREQUAL "WindowsPhone" AND NOT CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
@@ -440,7 +440,7 @@ if(X11_FOUND)
 	list(APPEND MEDIASTREAMER2_INCLUDE_DIRS ${X11_INCLUDE_DIRS})
 endif()
 if(XV_FOUND)
-	list(APPEND LINK_LIBS Xv)
+	list(APPEND LINK_LIBS Xv Xext)
 	list(APPEND MEDIASTREAMER2_INCLUDE_DIRS ${XV_INCLUDE_DIRS})
 endif()
 if(GLX_FOUND)
