--- mediastreamer-2.12.0/configure.ac.orig	2015-11-02 14:44:12.252105408 +0100
+++ mediastreamer-2.12.0/configure.ac	2015-11-10 19:53:44.207627872 +0100
@@ -82,7 +82,7 @@
 esac
 
 if test "$strictness" = "yes" ; then
-	STRICT_OPTIONS="$STRICT_OPTIONS -Werror"
+	STRICT_OPTIONS="$STRICT_OPTIONS -Werror -Wno-error=deprecated-declarations"
 	CFLAGS="$CFLAGS -fno-strict-aliasing"
 fi
 AC_SUBST(STRICT_OPTIONS)
