--- mediastreamer2-5.1.72/CMakeLists.txt.orig	2023-04-10 18:45:22.718792262 +0200
+++ mediastreamer2-5.1.72/CMakeLists.txt	2023-04-10 19:00:27.760555896 +0200
@@ -325,7 +325,13 @@ if(NOT APPLE AND NOT ANDROID AND OPENGL_
 		set(MS2_USE_OLD_OPENGL_PROTOTYPE TRUE)#Before 1.11, Glew have few different headers from OpenGL Specs
 	endif()
 endif()
-# TODO: Handle SDL option
+if(ENABLE_SDL)
+	find_package(SDL)
+	if(NOT SDL_FOUND)
+		message(WARNING "Could not find SDL library")
+		set(ENABLE_SDL OFF CACHE BOOL "Enable SDL video output." FORCE)
+	endif()
+endif()
 if(ENABLE_THEORA)
 	find_package(Theora)
 	if(NOT THEORA_FOUND)
@@ -498,6 +504,9 @@ if(GLEW_FOUND)
 	list(APPEND LINK_LIBS GLEW)
 	list(APPEND MEDIASTREAMER2_INCLUDE_DIRS ${GLEW_INCLUDE_DIRS})
 endif()
+if(SDL_FOUND)
+	list(APPEND LINK_LIBS SDL)
+endif()
 if(THEORA_FOUND)
 	list(APPEND LINK_LIBS theora)
 endif()
