--- mediastreamer2-4.5.22/src/CMakeLists.txt.orig	2021-06-24 23:55:04.000000000 +0200
+++ mediastreamer2-4.5.22/src/CMakeLists.txt	2021-07-07 06:15:26.781790314 +0200
@@ -586,7 +586,7 @@ endif()
 
 if(ENABLE_VIDEO)
   	install(FILES voip/nowebcamCIF.jpg
-		  DESTINATION ${CMAKE_INSTALL_DATADIR}/images
+		  DESTINATION ${CMAKE_INSTALL_DATADIR}/mediastreamer/images
 		  PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
 	)
 endif()
